analyzer/src/main/kotlin/Analyzer.kt

```diff
@@ -267,9 +271,29 @@ class Analyzer {
         return project.copy(scopes = filteredScopes.toSortedSet())
     }
 
+    private fun filterExcludedPackages(project: Project, excludedPackageIds: List<Identifier>): Project {
+        val filteredScopes = project.scopes.map { scope ->
+            val dependencies = scope.dependencies.map { pkgRef ->
+                pkgRef.traverse {
+                    if (it.id in excludedPackageIds) {
+                        it.copy(excluded = true)
+                    } else {
+                        it
+                    }
+                }
+            }
+
+            scope.copy(dependencies = dependencies.toSortedSet())
+        }
+
+        return project.copy(scopes = filteredScopes.toSortedSet())
+    }
+
     private fun filterUnreferencedPackages(projects: List<Project>, packages: SortedSet<CuratedPackage>)
             : SortedSet<CuratedPackage> {
-        val packageIdentifiers = projects.flatMap { it.scopes.flatMap { it.collectDependencyIds() } }.toSet()
+        val packageIdentifiers = projects.flatMap { project ->
+            project.scopes.flatMap { it.collectDependencyIds(includeExcluded = false) }
```

Comment:
`includeExcluded` sounds weird. We should re-think to generally use the term "ignored" instead of "excluded", so this would become `includeIgnored`. But that's out of the scope of this review.